unified_planning/engines/pddl_planner.py

```diff
@@ -69,19 +69,19 @@ def _get_cmd(self, domain_filename: str, problem_filename: str, plan_filename: s
         problem and write the plan on the file called plan_filename.'''
         raise NotImplementedError
 
-    def _plan_from_file(self, problem: 'up.model.Problem', plan_filename: str) -> 'up.plans.Plan':
-        '''Takes a problem and a filename and returns the plan parsed from the file.'''
+    def _plan_from_file(self, problem: 'up.model.Problem', plan_filename: str, renamings: Dict[str, str]) -> 'up.plans.Plan':
+        '''Takes a problem, a filename and a map of renamings and returns the plan parsed from the file.'''
         actions = []
         with open(plan_filename) as plan:
             for line in plan.readlines():
                 if re.match(r'^\s*(;.*)?$', line):
                     continue
                 res = re.match(r'^\s*\(\s*([\w?-]+)((\s+[\w?-]+)*)\s*\)\s*$', line)
                 if res:
-                    action = problem.action(res.group(1))
+                    action = problem.action(renamings[res.group(1)])
```

Comment:
For construction of the `renamings` map this should never happen, so a `keyError` would be a "nice" way to find a possible mistake in the code. We could add a default to just `res.group(1)` (The first draft was like that), but then i decided that this shouldn't happen.

Reply:
I was not suggesting to silence the error, but to raise a meaninful error instead of a `KeyError` :wink: 

("Errors should never pass silently" https://peps.python.org/pep-0020/)